components/work/WorkEdit.tsx

```diff
@@ -0,0 +1,143 @@
+import { FormEvent, PureComponent } from 'react';
+import { Container } from 'react-bootstrap';
+import { observable } from 'mobx';
+import { formToJSON } from 'web-utility';
+import { Button, Col, Form, Row } from 'react-bootstrap';
+
+import activityStore from '../../models/Activity';
+import { TeamWork } from '../../models/Team';
+import { NextRouter, withRouter } from 'next/router';
+import { FileUpload } from '../FileUpload';
+import { TeamWorkType } from '../../models/Team';
+const workTypes = [
+  { title: '网站', value: 'website' },
+  { title: '图片', value: 'image' },
+  { title: '视频', value: 'video' },
+  { title: 'Word', value: 'word' },
+  { title: 'PowerPoint', value: 'powerpoint' },
+  { title: 'PDF', value: 'pdf' },
+  { title: 'ZIP', value: 'zip' },
+];
+//TODO 当前表单组件用类组件的形式
+class WorkEdit extends PureComponent<{
+  work?: TeamWork;
+  router: NextRouter | any;
+}> {
+  store = activityStore
+    .teamOf(this.props.router.query.name)
+    .workOf(this.props.router.query.tid);
+
+  @observable
+  work: Partial<TeamWork> =
+    this.props.work || ({ type: TeamWorkType.WEBSITE } as TeamWork);
+  handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    const data = formToJSON(event.target as HTMLFormElement);
+    await this.store.updateOne(data);
+    const { router } = this.props;
+    await router.push(
+      `/activity/${router.query.name}/team/${router.query.tid}`,
+    );
+  };
+  changeWorkType = ({
+    currentTarget: { value },
+  }: FormEvent<HTMLInputElement>) =>
+    (this.work = { ...this.props.work, type: value as TeamWorkType });
+  render = () => {
+    const { work } = this.props;
+    return (
```

Comment:
```suggestion
    (this.work = { ...this.props.work, type: value as TeamWorkType });

  render = () => {
    const { work } = this;

    return (
```
`this.props` 不是**可观察对象**，没法根据组件内状态变化而更新